compiler/rustc_mir_transform/src/shim.rs

```diff
@@ -296,6 +297,31 @@ impl<'a, 'tcx> DropElaborator<'a, 'tcx> for DropShimElaborator<'a, 'tcx> {
     }
 }
 
+fn build_as_repr_shim<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId, self_ty: Ty<'tcx>) -> Body<'tcx> {
```

Comment:
@oli-obk can you take a second look at the changes in this file? LGTM, but you have more context here.

Reply:
👍 from me